crates/crates-io/lib.rs

```diff
@@ -393,6 +414,8 @@ impl Registry {
             headers.append("Content-Type: application/json")?;
         }
 
+        headers.append(&format!("User-Agent: cargo-cli"))?;
```

Comment:
This is not ideal. Cargo has its own [user-agent header](https://doc.rust-lang.org/nightly/cargo/reference/registry-web-api.html?highlight=user-agent#web-api) when communicating with crates.io. This kind of setting should be in `cargo` main crate, not `crates-io`. Also there is `curl::easy::Easy::useragent` you could use. `

Reply:
Yeah, I did some searching around and found that in `cargo::util::network::http` exists a function named `http_handle` that autoconfigures things like the user-agent, etc. based on the `gctx` (GloablContext) variable. I assume it is ok to use that?